### DISCUSSION TITLE
Enhanced Cover Page

### DISCUSSION THREAD

**Discussion by hugepanic** [Blackymas/NSPanel_HA_Blueprint#1852] [Ideas]
The cover functionality does not seem to work well with variable-angle covers ([Raffstore](https://de.wikipedia.org/wiki/Raffstore)).

![image](https://www.steg-platten.de/media/image/85/24/b3/Zetta-90-detail_600x600.png)

These elements can be located (up and down) and rotated.
For rotation, the motor is turned in the other direction.

E.g.: it takes 50s to go from up to down
to rotate the elements it takes 1s

--> In order to locate the cover to center and rotate to center you do the following:
move the motor UP for 25s
THEN
move the motor DOWN for 0.5s

### Current State:
The cover panel can locate the covers to a certain position

### Idea 1:
a "quick fix" would be a momentary switch on the GUI to rotate the motor in any direction, but only pretty short! --> e.g. 0.1s (that is possible with the Shelly2PM)  --> that way I can locate the covers to my desired location and then "fine tune" for the angle

### Idea 2:
A more "complete" approach would be to have a second slider that selects the angle of the elements.
This could then be combined with a sequence to allways "move" the covers and then "rotate".
--> this is the way it is done at other HomeAutomation systems (like the Busch-Jäger system).

As I am a new user for the NSpanel, please tell me if there are other solutions to this problem.
(my current workaround is a button on the buttons-page that drives the motor for 0.1s. This is not ideal as you have to switch from the cover panel to the buttons-page)

**Comment by edwardtfn** [comment]
Is this the same you are asking for?

- #1052

**Reply by hugepanic** [reply]
Yes!! That's the functionality I am missing!

**Reply by edwardtfn** [reply]
I will close this discussion here and we concentrate on that other thread, ok?

**Reply by hugepanic** [reply]
Yes!  Thanks
